desc/grid.py

```diff
@@ -48,24 +49,59 @@ class Grid(IOAble):
         "_inverse_rho_idx",
         "_inverse_theta_idx",
         "_inverse_zeta_idx",
-        "_num_rho",
-        "_num_theta",
-        "_num_zeta",
     ]
 
     def __init__(self, nodes, sort=True):
+        # Python 3.3 (PEP 412) introduced key-sharing dictionaries.
+        # This change measurably reduces memory usage of objects that
+        # define all attributes in their __init__ method.
         self._NFP = 1
         self._sym = False
         self._node_pattern = "custom"
-
         self._nodes, self._spacing = self._create_nodes(nodes)
-
         self._enforce_symmetry()
         if sort:
             self._sort_nodes()
-        self._find_axis()
-        self._count_nodes()
-        self._scale_weights()
+        self._axis = self._find_axis()
+        (
+            self._unique_rho_idx,
+            self._inverse_rho_idx,
+            self._unique_theta_idx,
+            self._inverse_theta_idx,
+            self._unique_zeta_idx,
+            self._inverse_zeta_idx,
+        ) = self._find_unique_inverse_nodes()
+        self._L = self.num_rho
+        self._M = self.num_theta
+        self._N = self.num_zeta
+        self._weights = self._scale_weights()
+
+    def _create_nodes(self, nodes):
```

Comment:
I just changed the order of the functions here to match the order they are called.